website/src/segment.tsx

```diff
@@ -106,7 +142,9 @@ export const AnnotatedForm = (
   if (showAnything) {
     const showSegments = p.levelOfDetail >= LevelOfDetail.Segmentation
     const translation = p.segment.englishGloss.join(", ")
-    const isSelected = p.wordPanelDetails.currContents?.id === p.segment.id
+    const isSelected =
+      (p.wordPanelDetails.currContents as Dailp.FormFieldsFragment)?.id ===
+      p.segment.id
```

Comment:
Overall, I felt I could pretty easily follow your code despite the fact that I'm not familiar with this area of the project. It took me a second to realize that a segment refers to either a word or a paragraph, but you make this pretty clear with comments, and you're very consistent with naming things throughout.